File: year_3/semester_6/PS/tsma/api/src/main/java/pt/isel/tsma/entity/dto/calendar/CalendarCreateDTO.java
package pt.isel.tsma.entity.dto.calendar;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.NonNull;
import pt.isel.tsma.entity.dto.shift.ShiftDurationDTO;
import pt.isel.tsma.entity.model.shift.Type;

import java.time.LocalDate;
import java.util.HashMap;
import java.util.List;

import static java.util.Collections.emptyMap;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class CalendarCreateDTO implements ICalendarDTO {

	@NonNull
	private String title;

	private String description;

	private LocalDate startDate;

	@NonNull
	private LocalDate endDate;

	@NonNull
	private List<SequenceDTO> sequences;

	private HashMap<Type, ShiftDurationDTO> shiftDurations = new HashMap<>(emptyMap());
}

File: year_3/semester_6/PS/tsma/spa/src/app/model/dtos/calendar/calendar-create-dto.ts
import {Type} from "../../type";

export class CalendarCreateDTO {
	title! 	     : string
	description  : string = ""

	startDate!   : string
	endDate!     : string

	sequences!	 : ShiftSequence[]

	constructor(title: string, description: string, startDate: string, endDate: string, sequences: ShiftSequence[]) {
		this.title = title;
		this.description = description;
		this.startDate = startDate;
		this.endDate = endDate;
		this.sequences = sequences;
	}
}

export class ShiftSequence {

	 memberId : number
	 sequence : Type[]

	constructor(memberId: number, sequence: Type[]) {
		this.memberId = memberId;
		this.sequence! = sequence;
	}

}
